homeassistant/components/iotty/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""The iotty integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+
+from iottycloud.device import Device
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.config_entry_oauth2_flow import (
+    OAuth2Session,
+    async_get_config_entry_implementation,
+)
+
+from . import coordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+type IottyConfigEntry = ConfigEntry[IottyConfigEntryData]
+
+
+@dataclass
+class IottyConfigEntryData:
+    """Contains config entry data for iotty."""
+
+    known_devices: set[Device]
+    coordinator: coordinator.IottyDataUpdateCoordinator
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
Is it not possible to change like this, here an everywhere else, so mypy knows the type of the config entry runtime data?
```suggestion
async def async_setup_entry(hass: HomeAssistant, entry: IottyConfigEntry) -> bool:
```